backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/InteropLowering.kt

```diff
@@ -482,6 +491,16 @@ private class InteropLoweringPart1(val context: Context) : BaseInteropIrTransfor
             )
         }
 
+        irClass.companionObject()?.let {
+            if (it.hasFields() ||
+                    it.getSuperClassNotAny()?.hasFields() ?: false) {
+                context.reportCompilationError(
+                        "Fields are not supported for Companion of subclass of ObjC type",
+                        currentFile, irClass
```

Comment:
Reporting the location of particular field (property?) would significantly improve user experience.

Reply:
Let's consider it as follow-up enhancement